garden_ai/globus_compute/containers.py

```diff
@@ -0,0 +1,89 @@
+from time import sleep
+from enum import Enum
+from rich.console import Console
+import logging
+
+from funcx import FuncXClient, ContainerSpec  # type: ignore
+from globus_sdk import GlobusAPIError
+from garden_ai.pipelines import Pipeline
+
+console = Console()
+logger = logging.getLogger()
+
+
+class ContainerBuildException(Exception):
+    """Exception raised when a container build request fails"""
+
+    pass
+
+
+class BuildStatus(str, Enum):
+    queued = "queued"
+    building = "building"
+    ready = "ready"
+    failed = "failed"
+
+
+def build_container(funcx_client: FuncXClient, pipeline: Pipeline) -> str:
+    name = str(pipeline.uuid)
+    cs = ContainerSpec(
+        name=name,
+        pip=[],  # pipeline.pip_requirements,
+        python_version="3.7",  # pipeline.python_version,
+        conda=[],  # pipeline.conda_requirements,
+    )
```

Comment:
Owen's PR that's landing soon will add these fields to the Pipeline. This is the only part I'll need to update to get this out of draft.